### ISSUE TITLE
Lambda RemovePermission throwing serialization error

### ISSUE THREAD

**Issue by sunnylbk** [localstack/localstack#1707]
- When attempting to remove a created lambda permission with the new fix from https://github.com/localstack/localstack/issues/1260, throws up `SerializationError`

Code being called:

```
func deleteLambdaPermission(ctx context.Context, destinationLambdaARN, permissionStatementID string) (err error) {
	lambdaClient := services.LambdaClient()
	lambdaRemovePermissionInput := &lambda.RemovePermissionInput{
		FunctionName: aws.String(destinationLambdaARN),
		StatementId:  aws.String(permissionStatementID),
	}
	_, err = lambdaClient.RemovePermissionWithContext(ctx, lambdaRemovePermissionInput)

	return
}
```

Checking for error in tests shows the following

```
 Expected
        <*awserr.requestError | 0xc0004a2180>: {
            awsError: {
                code: "SerializationError",
                message: "failed to unmarshal response error",
                errs: [
                    {
                        awsError: {
                            code: "UnmarshalError",
                            message: "failed decoding error message",
                            errs: [
                                {
                                    msg: "invalid character '<' looking for beginning of value",
                                    Offset: 1,
                                },
                            ],
                        },
                        bytes: [60, 33, 68, 79, 67, 84, 89, 80, 69, 32, 72, 84, 77, 76, 32, 80, 85, 66, 76, 73, 67, 32, 34, 45, 47, 47, 87, 51, 67, 47, 47, 68, 84, 68, 32, 72, 84, 77, 76, 32, 51, 46, 50, 32, 70, 105, 110, 97, 108, 47, 47, 69, 78, 34, 62, 10, 60, 116, 105, 116, 108, 101, 62, 52, 48, 52, 32, 78, 111, 116, 32, 70, 111, 117, 110, 100, 60, 47, 116, 105, 116, 108, 101, 62, 10, 60, 104, 49, 62, 78, 111, 116, 32, 70, 111, 117, 110, 100, 60, 47, 104, 49, 62, 10, 60, 112, 62, 84, 104, 101, 32, 114, 101, 113, 117, 101, 115, 116, 101, 100, 32, 85, 82, 76, 32, 119, 97, 115, 32, 110, 111, 116, 32, 102, 111, 117, 110, 100, 32, 111, 110, 32, 116, 104, 101, 32, 115, 101, 114, 118, 101, 114, 46, 32, 73, 102, 32, 121, 111, 117, 32, 101, 110, 116, 101, 114, 101, 100, 32, 116, 104, 101, 32, 85, 82, 76, 32, 109, 97, 110, 117, 97, 108, 108, 121, 32, 112, 108, 101, 97, 115, 101, 32, 99, 104, 101, 99, 107, 32, 121, 111, 117, 114, 32, 115, 112, 101, 108, 108, 105, 110, 103, 32, 97, 110, 100, 32, 116, 114, 121, 32, 97, 103, 97, 105, 110, 46, 60, 47, 112, 62, 10],
                    },
                ],
            },
            statusCode: 404,
            requestID: "",
            bytes: nil,
        }
    to be nil
```

**Comment by whummer** [comment]
Thanks for reporting @sunnylbk. Should be fixed in #1709 - can you please give it a try with the latest version?

**Comment by sunnylbk** [comment]
This works, thanks for fixing this!
